docs/adr/0039-use-react-container-component-abstraction.md

```diff
@@ -0,0 +1,93 @@
+# Use React Container Component Abstraction
+
+- Status: accepted
```

Comment:
I went ahead and preemptively marked this as accepted since I've had a decent portion of these problems documented in the (WIP still) [Settings architecture doc](https://docs.google.com/document/d/17dCwjxhxBKXDclkydmPHD91Pj3suIFihR6fZw6m126Q/edit#heading=h.3neqekw0gen1) for a month or two (I'll remove related stuff in it and link to this ADR instead), @dschom and I have collaborated on this through that doc and a few DMs, and the main part of the prototype has been up for a week and a half or so.